addons/medical_treatment/functions/fnc_checkItems.sqf

```diff
@@ -16,27 +16,22 @@ if (EGVAR(medical,convertItems) == 2) exitWith {};
 
 params ["_unit"];
 
-private _countFirstAidKit = [_unit, "FirstAidKit"] call EFUNC(common,getCountOfItem);
-_unit removeItems "FirstAidKit";
+// Item conversions are cached at preStart
+private _itemHash = uiNamespace getVariable QGVAR(replaceItems);
 
-private _countMedikit = [_unit, "Medikit"] call EFUNC(common,getCountOfItem);
-_unit removeItems "Medikit";
+private _replaceItem = {
+    private _itemCount = [_unit, _key] call EFUNC(common,getCountOfItem);
+    _unit removeItems _key;
 
-if (EGVAR(medical,convertItems) != 0) exitWith {};
+    if (EGVAR(medical,convertItems) != 0) exitWith {};
 
-for "_i" from 1 to _countFirstAidKit do {
-    _unit addItem "ACE_fieldDressing";
-    _unit addItem "ACE_packingBandage";
-    _unit addItem "ACE_morphine";
-    _unit addItem "ACE_tourniquet";
+    for "_i" from 1 to _itemCount do {
```

Comment:
for loop 🤢 
Could just do one for loop.
```sqf
{
    for "_j" from 1 to ((_x select 1) * _itemCount) do {
        _unit addItem (_x select 0);
    };
} forEach _value
```

Reply:
I knew I was missing an obvious way to simplify these loops! 😷